src/wallet/wallet2.cpp

```diff
@@ -779,16 +809,28 @@ bool wallet2::init(std::string daemon_address, boost::optional<epee::net_utils::
   return m_http_client.set_server(get_daemon_address(), get_daemon_login(), ssl);
 }
 //----------------------------------------------------------------------------------------------------
-bool wallet2::is_deterministic() const
+bool wallet2::is_deterministic(const boost::optional<epee::wipeable_string> &password)
 {
+  crypto::chacha_key key;
+  if (password)
```

Comment:
This (along with the corresponding decryption part) needs to check `password && m_ask_password && !m_rpc`. One issue from this bug: 1) Create a wallet with this patch, 2) `set ask-password 0`, 3) `set seed language` which returns "wallet is non-deterministic and has no seed"